### PR TITLE
Hubs/Scopes Merge 23 - Use new API for CRONS integrations

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-quartz/src/main/java/io/sentry/quartz/SentryJobListener.java [hunk 1]
```diff
@@ -49,13 +52,14 @@ public void jobToBeExecuted(final @NotNull JobExecutionContext context) {
       if (maybeSlug == null) {
         return;
       }
-      scopes.pushScope();
+      final @NotNull ISentryLifecycleToken lifecycleToken = scopes.pushIsolationScope();
```
Comment (adinauer) [inline sentry-quartz/src/main/java/io/sentry/quartz/SentryJobListener.java:55]: For this it likely makes sense to have it isolated.

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java [hunk 2]
```diff
@@ -86,7 +87,7 @@ public Object invoke(final @NotNull MethodInvocation invocation) throws Throwabl
       return invocation.proceed();
     }
 
-    scopes.pushScope();
+    final @NotNull ISentryLifecycleToken lifecycleToken = scopes.pushIsolationScope();
```
Comment (adinauer) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java:90]: Not 100% sure here, maybe we should make it configurable whether this isolates or not (`pushScope` vs. `pushIsolationScope`).

File: sentry/src/main/java/io/sentry/util/CheckInUtils.java [hunk 3]
```diff
@@ -30,12 +31,11 @@ public static <U> U withCheckIn(
       final @Nullable MonitorConfig monitorConfig,
       final @NotNull Callable<U> callable)
       throws Exception {
+    final @NotNull ISentryLifecycleToken lifecycleToken = Sentry.pushIsolationScope();
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/util/CheckInUtils.java:34]: Maybe this should also be configurable to use `pushScope` or `pushIsolationScope`.

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java (hunk 2 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java:90]: Default should probably be isolated.

File: sentry/src/main/java/io/sentry/util/CheckInUtils.java (hunk 3 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/util/CheckInUtils.java:34]: Default should probably also be isolated. Add overload with `isolated: true/false`, defaulting to true

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java (hunk 2 above)
Comment (markushi) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java:90]: Yeah makes sense! 
If any user code within `invocation.proceed();` performs changes via the static API, e.g. `Sentry.setTag()`, those will be written to the isolation scope which is created here, right?

Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java:90]: Yeah, static API and `configureScope` default to isolation scope for non Android.

File: sentry/src/main/java/io/sentry/util/CheckInUtils.java [hunk 4]
```diff
@@ -30,12 +31,11 @@ public static <U> U withCheckIn(
...
     final @NotNull IScopes scopes = Sentry.getCurrentScopes();
     final long startTime = System.currentTimeMillis();
     boolean didError = false;
 
-    // TODO fork instead
-    scopes.pushScope();
     TracingUtils.startNewTrace(scopes);
 
     CheckIn inProgressCheckIn = new CheckIn(monitorSlug, CheckInStatus.IN_PROGRESS);
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/util/CheckInUtils.java:41]: Could we use try-with-resource in the try-catch below?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java [hunk 5]
```diff
@@ -106,7 +107,7 @@ public Object invoke(final @NotNull MethodInvocation invocation) throws Throwabl
       CheckIn checkIn = new CheckIn(checkInId, monitorSlug, status);
       checkIn.setDuration(DateUtils.millisToSeconds(System.currentTimeMillis() - startTime));
       scopes.captureCheckIn(checkIn);
-      scopes.popScope();
+      lifecycleToken.close();
```
[the remark this answers is not in the document]
Comment (lbloder) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java:110]: Done in a follow-up PR

File: sentry-spring/src/main/java/io/sentry/spring/checkin/SentryCheckInAdvice.java [hunk 6]
```diff
@@ -89,7 +90,7 @@ public Object invoke(final @NotNull MethodInvocation invocation) throws Throwabl
       return invocation.proceed();
     }
 
-    scopes.pushScope();
+    final @NotNull ISentryLifecycleToken lifecycleToken = scopes.pushIsolationScope();
```
[the remark this answers is not in the document]
Comment (lbloder) [reply sentry-spring/src/main/java/io/sentry/spring/checkin/SentryCheckInAdvice.java:93]: Done in a follow-up PR

**Review by lbloder — APPROVED** [review]

**Pull request merged**